openeogeotrellis/deploy/batch_job.py

```diff
@@ -356,11 +356,13 @@ def run_job(
                 ml_model_metadata = result.get_model_metadata(str(output_file))
                 logger.info("Extracted ml model metadata from %s" % output_file)
             for name, asset in the_assets_metadata.items():
-                # TODO: test in separate branch
-                # if not asset.get("href").lower().startswith("s3:/"):
-                #     # fusemount could have some delay to make files accessible, so poll a bit:
-                #     asset_path = get_abs_path_of_asset(asset["href"], job_dir)
-                #     wait_till_path_available(asset_path)
+                href = asset["href"]
+                url = urlparse(href)
+                if url.scheme in ["", "file"]:
+                    file_path = url.path
+                    # fusemount could have some delay to make files accessible, so poll a bit:
+                    asset_path = get_abs_path_of_asset(file_path, job_dir)
+                    wait_till_path_available(asset_path)
```

Comment:
about `wait_till_path_available` : I think it's overkill to implement exponential back-off here as the "exist" check is pretty cheap.
Now with the exponential backoff, in the last try there is a sleep of 2 minutes (128s), which is very long. I would just try every 5 seconds up to 1 or 2 minutes in total

Reply:
The exists check could make a request to S3 if it is a mount. So multiple jobs doing a query at the same rate might impact the S3 connection.
Only in the test suite it refers to local files I think

Reply:
true, an "exist" over S3 is probably not that cheep,  
but still, doing exponential sleep with base 2 (8s, 16s, 32s, 64s and 128s at the moment)  feels a bit pessimistic

Reply:
Changed! I am merging already to test on staging again